creator-node/src/middlewares.js

```diff
@@ -98,6 +100,41 @@ async function ensurePrimaryMiddleware (req, res, next) {
   next()
 }
 
+/** Blocks writes if node has used over `maxStorageUsedPercent` of its capacity. */
+async function ensureStorageMiddleware (req, res, next) {
+  // Get storage data and max storage percentage allowed
+  const [storagePathSize, storagePathUsed] = await getMonitors([
+    MONITORS.STORAGE_PATH_SIZE,
+    MONITORS.STORAGE_PATH_USED
+  ])
+
+  const maxStorageUsedPercent = config.get('maxStorageUsedPercent')
+
+  // Check to see if CNode has enough storage
+  let hasEnoughStorage = hasEnoughStorageSpace({
+    storagePathSize,
+    storagePathUsed,
+    maxStorageUsedPercent
+  })
+
+  if (hasEnoughStorage) {
+    next()
+  } else {
+    const errorMsg = `Node is reaching storage space capacity. Current usage=${(100 * storagePathUsed / storagePathSize).toFixed(2)}% | Max usage=${maxStorageUsedPercent}%`
+    req.logger.error(errorMsg)
+    return sendResponse(
+      req,
+      res,
+      errorResponseBadRequest(
```

Comment:
this should be returning 5xx not 4xx since this is server error not client error. lets use errorResponseServerError instead